src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,176 @@
+
+    ActorInfoContributorLifetimeExtension(final OrbitRemoteReferenceService orbitRemoteReferenceService,
+                                          final LifetimeExtension springLifetimeExtension)
+    {
+        this.orbitRemoteReferenceService = orbitRemoteReferenceService;
+        this.springLifetimeExtension = springLifetimeExtension;
+    }
+
+    @Override
+    public <T> T newInstance(final Class<T> concreteClass)
+    {
+        // In case Orbit chooses this extension as the "primary" LifetimeExtension, we need to redirect to the
+        // SpringLifetimeExtension for constructing new instances
+        return springLifetimeExtension.newInstance(concreteClass);
+    }
+
+    @Override
+    public Task<?> preActivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::add);
+    }
+
+    @Override
+    public Task<?> preDeactivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::remove);
+    }
+
+    private Task<?> doIfInfoContributor(final AbstractActor<?> actor,
+                                        final Consumer<ActorInfoContributorReference> synchronizedAction)
+    {
+        if (actor instanceof InfoContributor)
+        {
+            ActorInfoContributorReference actorInfoContributorReference = new ActorInfoContributorReference(actor);
+            synchronized (actorInfoContributorReferences)
+            {
+                synchronizedAction.accept(actorInfoContributorReference);
+            }
+        }
+        return Task.done();
+    }
+
+    @Override
+    public void contribute(final Info.Builder builder)
+    {
+        createActorInfoObject().ifPresent(actorInfoObject -> builder.withDetail("actors", actorInfoObject));
+    }
+
+    private Optional<?> createActorInfoObject()
+    {
+        Set<ActorInfoContributorReference> actorInfoContributorReferencesCopy;
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferencesCopy = actorInfoContributorReferences.stream()
+                    .map(ActorInfoContributorReference::new)
+                    .collect(Collectors.toSet());
+        }
+        Collection<ActorInfoContributorReference> expiredReferences = new ArrayList<>();
+        Map<String, Map<String, Object>> actorInfoObject = new HashMap<>();
+        actorInfoContributorReferencesCopy.forEach(actor ->
+        {
+            InfoContributor contributor = actor.reference.get();
+            if (contributor == null)
+            {
+                expiredReferences.add(actor);
+            }
+            else
+            {
+                final Info.Builder builder = new Info.Builder();
+                contributor.contribute(builder);
+                actorInfoObject.putIfAbsent(actor.name, new HashMap<>());
+                actorInfoObject.get(actor.name).put(actor.identity, builder.build().getDetails());
```

Comment:
My concern is that we are almost going down the path of just exposing a significant number of activated Actor states by providing this structure. This is why I would like to understand more around the use case that this capability is trying to satisfy. If that is the case then it would make more sense to have a `/actors` endpoint where you have a structure like what you described so you can see a per actor instance state (keyed by the actor identifier).